apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationDetails.tsx

```diff
@@ -60,8 +63,13 @@ const ApplicationDetails = ({
   );
 
   const filteredEntities = useMemo(() => {
-    return entities.filter((e) => entity.name === e.name);
-  }, [entities, entity.name]);
+    return allEntities.filter(
+      (e) =>
+        entity.name === e.name &&
+        (!onlyInstalledVersions ||
+          installedModels.map(({ id }) => id).includes(e.id)),
```

Comment:
`installedModels.map(({ id }) => id)` should be moved out of loop to avoid doing it in each iteration